deblur/workflow.py

```diff
@@ -7,20 +7,35 @@
 # -----------------------------------------------------------------------------
 
 
-def trim_seqs(seqs_fp, trim_len):
-    """Step 1: trim FASTA sequences to specified length"""
-    pass
+from os.path import splitext
+from bfillings.vsearch import vsearch_dereplicate_exact_seqs
 
 
-def dereplicate_seqs(seqs_fp):
-    """Step 2a: dereplicate FASTA sequences using VSEARCH"""
+def trim_seqs(seqs_fp, trim_len):
+    """Step 1: trim FASTA sequences to specified length"""
     pass
 
 
-def remove_singletons_seqs(seqs_fp):
-    """Step 2b: remove singletons from dereplicated FASTA
-       file using VSEARCH"""
-    pass
+def dereplicate_seqs(seqs_fp,
+                     output_fp,
+                     min_size=2):
+    """Step 2a: dereplicate FASTA sequences and remove
+       singletons using VSEARCH
+
+    Parameters
+    ----------
+    seqs_fp : string
+        filepath to FASTA sequence file
+    output_fp : string
+        filepath to dereplicated FASTA file
+    """
+    log_name = "%s.log" % splitext(output_fp)[0]
```

Comment:
the log file will be called XXX.YYY.log

I can change it to add '.log' to the end of the input file name, without assuming the last string after '.' is the extension ..